service/history/reset/resetter.go

```diff
@@ -285,7 +285,7 @@ func (r *workflowResetterImpl) persistToDB(
 	if err != nil {
 		return err
 	}
-	resetHistorySize, err := resetWorkflow.GetContext().PersistFirstWorkflowEvents(resetWorkflowEventsSeq[0])
+	resetHistorySize, err := resetWorkflow.GetContext().PersistNonFirstWorkflowEvents(resetWorkflowEventsSeq[0])
```

Comment:
I am not following here. Why do we want to use persist non first workflow events for the reset sequence[0]?

Reply:
If you are asking why PersistNonFirstWorkflowEvents instead of PersistFirstWorkflowEvents -- because PersistFirstWorkflowEvents will try to insert another record for "history_tree", which is not supposed to do. Because "ForkBranch" already done that. Redoing it here will cause potential issues. It works with Cassandra because it's using upsert. Theoretically we should use insert, but inset is too expensive for Cassandra.

If you are asking why resetWorkflowEventsSeq[0] not resetWorkflowEventsSeq[1 or others], I think that's some details for the workflow interface. We should do an assertion here to make sure len(resetWorkflowEventsSeq) == 1.

Reply:
Added an assertion